File: Python/labs/lab5/task_3_1/buttons.py
from abc import ABC, abstractmethod
from sizes import ButtonSize


class ButtonType(ABC):
    def __init__(self, button_size: ButtonSize):
        self._button_size = button_size

    @abstractmethod
    def draw_button(self):
        pass


class RadioButton(ButtonType):
    def __init__(self, button_size: ButtonSize):
        super(RadioButton, self).__init__(button_size)

    def draw_button(self):
        print(f'Drawing radio button [{self._button_size.get_size()}]')


class CheckboxButton(ButtonType):
    def __init__(self, button_size: ButtonSize):
        super(CheckboxButton, self).__init__(button_size)

    def draw_button(self):
        print(f'Drawing checkbox button [{self._button_size.get_size()}]')


class DropdownButton(ButtonType):
    def __init__(self, button_size: ButtonSize):
        super(DropdownButton, self).__init__(button_size)

    def draw_button(self):
        print(f'Drawing dropdown button [{self._button_size.get_size()}]')


class SimpleButton(ButtonType):
    def __init__(self, button_size: ButtonSize):
        super(SimpleButton, self).__init__(button_size)

    def draw_button(self):
        print(f'Drawing simple button [{self._button_size.get_size()}]')

File: Cpp/lab08_Bridge/task_2_2 - solution/src/buttons.cpp
#include <iostream>

#include "buttons.hpp"


//Button
Button::Button(ControlSize* controlSize)
: controlSize(controlSize) {}

void Button::draw()
{
  this->controlSize->applySize();
}


//CheckboxButton
CheckboxButton::CheckboxButton(ControlSize* controlSize)
: Button(controlSize) {}

void CheckboxButton::draw()
{
  this->Button::draw();
  std::cout << "Drawing a checkbox button." << std::endl;
}


//DropdownButton
DropdownButton::DropdownButton(ControlSize* controlSize)
: Button(controlSize) {}

void DropdownButton::draw()
{
  this->Button::draw();
  std::cout << "Drawing a dropdown button." << std::endl;
}


//RadioButton
RadioButton::RadioButton(ControlSize* controlSize)
: Button(controlSize) {}

void RadioButton::draw()
{
  this->Button::draw();
  std::cout << "Drawing a radio button." << std::endl;
}
